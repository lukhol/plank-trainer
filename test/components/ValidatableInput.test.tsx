import React from 'react';
import ValidatableInput,  {ValidatableInputProps } from '../../src/components/ValidateableInput';
import renderer from 'react-test-renderer';

describe(' ------ ValidatableInput ------ ', () => {
    const getComponent = (props: ValidatableInputProps) => {
        return (
            renderer.create(
                <ValidatableInput
                    {...props}
                />
            )
        )
    };

    it('renders correctly with defaults and valid state', () => {
        const validatableInput = getComponent({
            value: 'Some value',
            onChangeText: (text: string) => {},
            isValid: true
        });

        expect(validatableInput.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with defaults and invalid state', () => {
        const validatableInput = getComponent({
            value: 'Some value',
            onChangeText: (text: string) => {},
            isValid: false
        });

        expect(validatableInput.toJSON()).toMatchSnapshot();
    });
});