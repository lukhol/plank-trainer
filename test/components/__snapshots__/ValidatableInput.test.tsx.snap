// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` ------ ValidatableInput ------  renders correctly with defaults and invalid state 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "red",
      }
    }
    underlineColorAndroid="transparent"
    value="Some value"
  />
  <Text
    style={
      Object {
        "color": "red",
        "marginHorizontal": 4,
      }
    }
  />
</View>
`;

exports[` ------ ValidatableInput ------  renders correctly with defaults and valid state 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#89f450",
      }
    }
    underlineColorAndroid="transparent"
    value="Some value"
  />
</View>
`;
